kbs/docs/plugins/pkcs11.md

```diff
@@ -0,0 +1,33 @@
+# PKCS11 plugin
+
+PKCS11 is a broad specification supporting many cryptographic operations, mainly applicable to Hardware Security Modules (HSMs). Trustee implements a subset of features that can be of use to attested clients.
+
+## Resource Storage
+
+The PKCS11 plugin can store plaintext resources in an HSM. Since the KBS expects resources to be in plaintext, we store these resources in the HSM as secret keys of the generic secret type. This storage interface will provision resources to the HSM in the expected way when a user uploads a resource to the KBS. The user must simply specify the location of an initialized HSM slot. A user can fetch resources using the same semantics as a resource backend.
```

Comment:
Just out of curiosity, why are resources stored as keys?  Wouldn't say private data objects be a better match in terms of semantics?  (Apologies if I'm making no sense, I don't have experience with PKCS#11.)

Reply:
IIUC although called "secret keys", they can't be used as "keys" for encryption, decryption, etc. They can be thought of a "secret objects" in that sense.

See [here](https://www.cryptsoft.com/pkcs11doc/v220/group__SEC__12__7__2__GENERIC__SECRET__KEY__OBJECTS.html).